ONSdigital/dp-frontend-search-controller
# Feature/implement new compact search

### What

- Removed duplicated effort by having inbuilt 'compact-search' rather than instantiating via `dp-renderer`
- Instantiated `dp-renderer` compact search and associated model binding
- Removed unnecessary view toggle `KeywordFilterEnabled` as the keyword search is on all pages
- Changed duplicate `id`
- Removed unnecessary `fieldset` for one field
- Updated go docker image
- Update `dp-design-system`

### How to review

- Sense check
- Image review
- Try the changes for yourself
  - Pull this branch and run app `make debug ENABLE_REWORKED_DATA_AGGREGATION_PAGES=true`
  - Run `dp-design-system`
  - Port forward the api router to sandbox `dp ssh sandbox web 1 -p 23200:localhost:10800`
  - Go to the following routes:
     - http://localhost:25000/alladhocs
     - http://localhost:25000/datalist
     - http://localhost:25000/publishedrequests
     - http://localhost:25000/staticlist
     - http://localhost:25000/topicspecificmethodology
     - http://localhost:25000/timeseriestool
     - http://localhost:25000/publications
     - http://localhost:25000/allmethodologies
   - Check the keyword search field still works

#### Image
<img width="405" alt="new keyword search input" src="https://github.com/ONSdigital/dp-frontend-search-controller/assets/19624419/1f23102b-ba9d-486c-9265-409cdf5ada35">

### Who can review

!me
